src/common.rs

```diff
@@ -64,6 +64,17 @@ pub fn ucs2_as_ascii_length(input: *const u16) -> usize {
     len
 }
 
+pub fn ascii_length(input: &str) -> usize {
+    let mut len = 0;
+    for c in input.chars() {
+        if c == '\0' {
+            break;
+        }
+        len += 1;
+    }
+    len
+}
```

Comment:
Sort of surprised you need to need to do this, is there not a core::str::len ?

Reply:
Or is this slightly different? This is more like `strlen()`?

Reply:
Actually, `core::str::len()` returns size of the array when given str is converted from [u8; N].

Let's run the following snippet.
```rust
fn main() {
    let s = "foo";
    let mut a = [0_u8; 8];
    a[0] = b'b';
    a[1] = b'a';
    a[2] = b'r';
    let u = core::str::from_utf8(&a).unwrap();
    println!("s.len(): {}", s.len());
    println!("u.len(): {}", u.len());
    println!("ascii_length(u): {}", ascii_length(u));
}
```
We got:

```
s.len(): 3
u.len(): 8
ascii_length(u): 3
```

`s` is string literal, and len() returns 3 as expected. But `u` type is str too, but it is from [u8; N], and len() returns 8. This is different behavior from C's `strlen()`.

Note: We can avoid this pitfall by calling `u.trim_matches(char::from(0));` before `len()` call. I'll update to use this.